Assets/Scripts/Audio/AudioDistanceController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+/// <summary>
+/// This script is for making audio player louder/quieter depending on the distance to the audio
+/// source. 
+/// 
+/// Make sure to look at the object's parent and assaign the correct localScale.
+/// </summary>
+
+
+public class AudioDistanceController : MonoBehaviour {
+
+    public float maxVolume;
+
+    //The radius of the circle collider is determined by the local scale (For example if 
+    // the parent container was scaled to 0.5 and the radius was 15, the actuall global radius 
+    // would only be 7.5. 
+    public float localScale;
+    private CircleCollider2D audioRangeCollider;
+    private AudioSource audioSource;
+    private List<GameObject> dolls = new List<GameObject>();
+
+    // Use this for initialization
+    void Start() {
+        audioRangeCollider = GetComponent<CircleCollider2D>();
+        audioSource = GetComponent<AudioSource>();
+    }
+
+    void Update() {
+        //Check if the list of dolls includes the player doll, else the volume of the object should be 0
+        GameObject player = dolls.Find(x => x.CompareTag("Player"));
```

Comment:
You can't make this check to when the doll lists change because the doll list only changes when a doll enters the collider. This causes an issue where if you go into the collider and then switch out of the collider, you will still hear the audio.